onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -365,12 +407,20 @@ Status CommonSubexpressionElimination::ApplyImpl(Graph& graph, bool& modified, i
     for (const NodeArg* input_def : node->InputDefs()) {
       auto it = equivalence_classes.find(input_def);
       if (it == equivalence_classes.end()) {
+        const ONNX_NAMESPACE::TensorProto* initializer = nullptr;
+        
+        if(graph.IsInitializedTensor(input_def->Name())) {
```

Comment:
Need to also check that the initializer cannot be overridden ("initializers as inputs"). See `graph.GetOverridableInitializers()`.